test/main.hs

```diff
@@ -19,3 +20,9 @@ main = do
         ]
     let str = $(embedStringFile "test/sample/foo") :: String
     filter (/= '\r') str @?= "foo\n"
+
+    let mbs = $(embedFileIfExists "test/sample/foo")
+    mbs @?= Just "foo\r\n"
```

Comment:
This is likely going to have some cross-platform issues with CRLF, see the `filter` above. We can wait to see if CI succeeds or fails.